tests/utils/test_cache_utils.py

```diff
@@ -177,9 +186,118 @@ def _random_kvs(config):
         self.assertTrue(cached_values.shape == (1, 1, 10, 128))
 
 
-@require_torch_accelerator
 class CacheIntegrationTest(unittest.TestCase):
-    """Cache tests that require loading models"""
+    """Fast cache integration tests that share the same small model"""
+
+    @classmethod
+    def setUpClass(cls):
+        # Load once and reuse across tests
+        cls.tokenizer = AutoTokenizer.from_pretrained("HuggingFaceTB/SmolLM2-135M-Instruct", padding_side="left")
+        cls.model = AutoModelForCausalLM.from_pretrained(
+            "HuggingFaceTB/SmolLM2-135M-Instruct", device_map="auto", torch_dtype=torch.float16
+        )
+        cls.model.config.sliding_window = 256  # hack to enable the use of caches with sliding windows
+
+    @parameterized.expand(TEST_CACHE_IMPLEMENTATIONS)
+    def test_cache_batched(self, cache_implementation):
+        """Sanity check: caches' `.update` function expects batched inputs"""
+        if cache_implementation == "quantized" and not is_optimum_quanto_available():
+            self.skipTest("Quanto is not available")
+
+        EXPECTED_GENERATION = ["A sequence: 1, 2, 3, 4, 5, 6, 7, 8,", "A sequence: A, B, C, D, E, F, G, H"]
+
+        inputs = self.tokenizer(
+            ["A sequence: 1, 2, 3, 4, 5", "A sequence: A, B, C"], padding=True, return_tensors="pt"
+        )
+        inputs = inputs.to(self.model.device)
+
+        gen_out = self.model.generate(
+            **inputs,
+            do_sample=False,
+            max_new_tokens=10,
+            return_dict_in_generate=True,
+            cache_implementation=cache_implementation,
+            disable_compile=True,
+        )
+        # Sanity check: a cache was used
+        self.assertIsInstance(gen_out.past_key_values, Cache)
+        # Confirm that the output matches expectations
+        decoded = self.tokenizer.batch_decode(gen_out.sequences, skip_special_tokens=True)
+        self.assertListEqual(decoded, EXPECTED_GENERATION)
+
+    @parameterized.expand(TEST_CACHE_IMPLEMENTATIONS)
+    def test_cache_beam_search(self, cache_implementation):
+        """
+        Sanity check: caches' `reorder_cache` is operational. We can confirm this by looking at the beam indices
+        (an output sequence contains multiple beam indices).
+        """
+        if cache_implementation == "quantized" and not is_optimum_quanto_available():
+            self.skipTest("Quanto is not available")
+
+        if cache_implementation == "offloaded_hybrid_chunked":
+            # TODO (joao, cyril): something is off with `offloaded_hybrid_chunked` aka `OffloadedHybridCache`: the
```

Comment:
I don't think `offloaded_hybrid_chunked` + beam_search is worth the dive for now 🤔

Reply:
nope agree with you!